### ISSUE TITLE
minus-circle not working

### ISSUE THREAD

**Issue by flutefreak7** [spyder-ide/qtawesome#97]
I'm on Windows 10 with PyQt5.

The following demo:

    from PyQt5 import QtWidgets, QtGui, QtCore
    import qtawesome as qta

    app = QtWidgets.QApplication([])

    icon = qta.icon('fa5s.minus-circle')

    widget = QtWidgets.QWidget()
    layout = QtWidgets.QVBoxLayout()
    button = QtWidgets.QPushButton()
    button.setFixedHeight(100)
    button.setIconSize(QtCore.QSize(100, 100))
    button.setIcon(icon)
    layout.addWidget(button)

    widget.setLayout(layout)

    widget.show()

    app.exec_()

Gives me this:
![image](https://user-images.githubusercontent.com/2966015/45965378-e8e0c880-bfed-11e8-896b-c931245b2707.png)

**Comment by flutefreak7** [comment]
Hmmm, there is more going on... `user-minus` is giving me an icon, just the wrong one...

![image](https://user-images.githubusercontent.com/2966015/45965515-45dc7e80-bfee-11e8-8c18-278bb85e084f.png)

**Comment by darkvertex** [comment]
@ccordoba12 yep...

(My Linux OS skin is dark, so forgive the dark screenshots.)

I'm on CentOS Linux at the moment. I tried the first code snippet with `fa5s.minus-circle` (and color "red" so you can see it with my dark theme) and I got:

![image](https://user-images.githubusercontent.com/941331/46024372-a7672080-c0b4-11e8-9127-78e0ae1a8190.png)
which looks correct with [what the FontAwesome site says `minus-circle` is.](https://fontawesome.com/icons/minus-circle?style=solid)

I tried with `fa5s.user-minus` and I get [the good icon](https://fontawesome.com/icons/user-minus?style=solid) still:

![image](https://user-images.githubusercontent.com/941331/46024494-e85f3500-c0b4-11e8-993d-8524ba39437d.png)

I gather from the style of your window in your screenshot that you're in Windows. (Am I right?) It should work the same way in theory.

I haven't had a chance to try to reproduce the issue on Windows myself yet but I'll try when I'm home on my laptop. (I only have Linux at work.)

Any other details about your environment you'd like to share? It may be helpful to know a few more things: What Windows version are you on, 8 or 10? What PyQt5 version do you have at the moment? On what Python version are you?

**Comment by flutefreak7** [comment]
Windows 10
Python 3.6.4 through Anaconda
Qt 5.11.1
PyQt: 5.11.2

I've done some investigating through the qtawesome source and added the following to turn off "Font Merging" that Qt does when it can't find the requested character and falls back to a system font (which is the reason I was seeing unusual glyphs for some characters).

        font = iconic.font(prefix, draw_size)
        font.setStyleStrategy(QFont.NoFontMerging)
        painter.setFont(font)

When I do that I get no icon at all for many of the unicode characters in fontawesome's fonts.

I created a ListWidget of all the solid icons with the above modification.  Below is a screenshot of the first 1/3 or so:

![image](https://user-images.githubusercontent.com/2966015/46099218-d1d7dd00-c18b-11e8-9ad7-921845775c52.png)

This lead me to the realization that it is showing icons from the Regular style instead of the Solid style. In the Qt Font database these two fonts have the same family name "Font Awesome 5 Free":
![image](https://user-images.githubusercontent.com/2966015/46099666-ff715600-c18c-11e8-80c6-3de9ef5ca430.png)

This is dealt with by them having different Style Names:
![image](https://user-images.githubusercontent.com/2966015/46099708-1adc6100-c18d-11e8-8fcd-3ae598104826.png)

...but apparently, despite attempting to get the solid style by name, the resulting QFont is still attempting to look up the characters (the uniode codepoints from the Solid json character map) in the Regular font.  A few of the codepoints overlap (like how Address Book is U+F2B9 in both), yielding the incorrect regular version of the desired icon, but for most, Qt goes looking up what it thinks is a similar character in a system font.

I did request the available style names for the font with QFontDatabase and get Regular and Solid:
![image](https://user-images.githubusercontent.com/2966015/46100151-59265000-c18e-11e8-886e-c6f57caf1d9c.png)

Below is my demo code:

    from PyQt5 import QtWidgets, QtGui, QtCore
    import qtawesome as qta
    from six import unichr
    import json

    app = QtWidgets.QApplication([])

    icon = qta.icon('fa5s.angle-right')

    filename = r'C:\Anaconda3\Lib\site-packages\qtawesome\fonts\fontawesome5-solid-webfont-charmap.json'

    def hook(obj):
        result = {}
        for key in obj:
            result[key] = unichr(int(obj[key], 16))
        return result

    with open(filename, 'r') as f:
        charmap = json.load(f, object_hook=hook)

    widget = QtWidgets.QListWidget()

    for name, char in charmap.items():
        icon = qta.icon('fa5s.{}'.format(name))
        item = QtWidgets.QListWidgetItem()
        item.setIcon(icon)
        item.setText(name)
        widget.addItem(item)

    widget.show()

    app.exec_()

**Comment by ccordoba12** [comment]
@flutefreak7, thanks for going deep with this. It seems the new FontAwesome 5 support won't work on Windows, sorry.

**Comment by flutefreak7** [comment]
There are clearly lots of workarounds to get around this if it's a Windows 10 QFont issue:

- Rename the font family of the TTF files so that they are all unique - https://github.com/chrissimpkins/fontname.py appears to work fine for this
- Somehow use the font weight of regular and solid to differentiate them... the original files from fontawesome say the weights are 400 and 900, so perhaps one can get QFont to play along by specifying a weight
- Dynamically load the font file you need and/or unload the conflicting one - I tried this with my demo of all the icons and it was prohibitively slow of course.
- Just use the SVG files instead of the TTF files - that'd obviously be a completely different approach that would require different code and wouldn't work for any icon sets that don't provide SVG's

I came up with a solution that works for me that does the TTF font-family rename dynamically within the _instance() function if the user is on Windows and has fontTools installed.  I also had to temporarily disable the hash check for the solid file.

This is rough, but perhaps could inspire a cleaner fix or help someone else with the same problem.

Sorry I'm not great at Git, so here's a zip file of my changes:
[windows_fix.zip](https://github.com/spyder-ide/qtawesome/files/2421756/windows_fix.zip)

**Comment by flutefreak7** [comment]
Tried using `setWeight` to get the correct Style and that isn't working.

**Comment by jules-ch** [comment]
@flutefreak7 
I had the same issue on Windows 7 & PyQt 5.10.1 :

It seems setWeight does the trick : 
```
    def font(self, prefix, size):
        """Return a QFont corresponding to the given prefix and size."""
        font = QFont(self.fontname[prefix])
        if prefix[-1] == 's': # solid style
            font.setWeight(900)
        font.setPixelSize(size)
        return font
```

as a result : 

![image](https://user-images.githubusercontent.com/43635101/46131346-b0362e80-c23b-11e8-8fd0-c0b2a4f9f703.png)

**Comment by darkvertex** [comment]
@jules-ch: Nice! Wanna go ahead and make a PR for this little fix?

@ccordoba12: It seems harmless to me but do you think there are any repercussions to setting the font weight for `fa5s`?

**Comment by jules-ch** [comment]
Tried this at home on Windows 10 & PyQt 5.11.2 and setWeight is not working as @flutefreak7 said.
Don't know how we could have a proper workaround.

**Comment by darkvertex** [comment]
I think I'd prefer renaming the fonts with that renamer tool. It sounds like the safest choice.

**Comment by flutefreak7** [comment]
Just to add to the confusion, according to the docs, Qt uses a 0-99 weight system rather than the system used by... CSS, typography people, and everyone else, so I'm not sure what value is even expected for that to work. I do recall someone on Mac in a Stack Overflow post having issues with setWeight for complex fonts with lots of different weights.

**Comment by flutefreak7** [comment]
When I check font.weight() for the 2 fonts I get 50 for both (which should be the default), so at least in my environment that doesn't appear to work as a way to differentiate the two.

**Comment by ccordoba12** [comment]
@flutefreak7, @jules-ch, we're welcome to take the solution you think is the best one because fixing this is not a priority for us.

**Comment by darkvertex** [comment]
@flutefreak7 @jules-ch [I made a PR using fontname.py to edit the font families](https://github.com/spyder-ide/qtawesome/pull/98), could you give it a try and tell me if it works better for you guys?

**Comment by darkvertex** [comment]
Fixed in **[v0.5.1](https://github.com/spyder-ide/qtawesome/releases/tag/v0.5.1)** - enjoy! 🎉
